stumpy/motifs.py

```diff
@@ -126,6 +126,7 @@ def _motifs(
             atol=atol,
             query_idx=candidate_idx,
             T_subseq_isconstant=T_subseq_isconstant,
+            Q_subseq_isconstant=T_subseq_isconstant[:, [candidate_idx]],
```

Comment:
Why not just `T_subseq_isconstant[:, candidate_idx],`?

Is the extra square brackets necessary?

Reply:
I think it is necessary as other wise we are passing 1D boolean array that corresponds to a multi-dim time series, and hence each 1D Q (of multi-dim Q) corresponds to a single boolean value. However, this does not work as we need to provide boolean array (of shape `(1,)`) and not just boolean value. 

It might be a good idea to add support for boolean value in function `core.rolling_isconstant`. So, when we pass a boolean value to this function, it will be converted to 1d array (of length 1).

Reply:
From what I can tell, this:

```
T_subseq_isconstant[:, candidate_idx]
```

returns a 1D array with a single element. When this gets passed into `match`, the `match` function will check the dimensions of `Q_subseq_isconstant` and makes sure it is appropriately 2-dimensional. Maybe I'm missing something?

Reply:
I got confused by the dimension and I explained it in the wrong way. 

---

In `motifs.py`, we add a new axis to `T` and pass it to `_motifs`. (So, the input `T` in function `_motifs` is 2D. We may add it to the description of `T` in the docstring of `_motifs` that says it has the shape of `(1, len(T))`.

In function `_motifs`, we have:  
```
Q = T[:, candidate_idx : candidate_idx + m] 
```
which means `Q` is 2D. 

Note that `Q` and `Q_subseq_isconstant` that are being pass to function `match` must have the same dimension. Otherwise, we would  get an error when we do::

```
Q_subseq_isconstant = core.rolling_isconstant(Q, m, Q_subseq_isconstant)
```

Please note that the following if-block appears later in the code.
```
if Q_subseq_isconstant.ndim == 1:
    # add new dim
```

I hope that I did not make any mistage this time :smile:

Reply:
Okay. I think my biggest issue is that it's not obvious to me what the output dimensions would be with that syntax. What about:

```
Q_subseq_isconstant=np.atleast_2d(T_subseq_isconstant[:, candidate_idx])
```

Reply:
I trust your judgement as you have better insight. I think `np.atleast_2d` should be clear enough,